helpers.js

```diff
@@ -54,41 +78,56 @@ function handleArray(field) {
 }
 
 function handleArea(field) {
-  const widgets = Object.keys(field.options.widgets).reduce((widgetAcc, widgetCur) => {
-    if (widgetTypes[widgetCur]) {
-      const newWidgetName = widgetTypes[widgetCur];
-      widgetAcc[newWidgetName] = field.options.widgets[widgetCur];
-    }
-    return widgetAcc;
-  }, {});
+  if (field.type === 'singleton') {
+    return {
+      label: field.label,
+      type: 'area',
+      options: {
+        widgets: getWidgets(field.options, field.widgetType),
+        max: 1
+      }
+    };
+  }
 
   return {
     ...field,
     options: {
-      widgets
+      widgets: Object.keys(field.options.widgets).reduce((widgetAcc, widgetCur) => {
```

Comment:
No support yet for forms widgets. Everything else seems ok to me.

Reply:
I've created 2 issues for missing mapping.

* PRO-3321 https://linear.app/apostrophecms/issue/PRO-3321/add-support-for-forms-widgets
* PRO-3322 https://linear.app/apostrophecms/issue/PRO-3322/add-support-for-advanced-permission-group